shared/src/main/scala/squants/Quantity.scala

```diff
@@ -306,5 +306,10 @@ abstract class Quantity[A <: Quantity[A]] extends Serializable with Ordered[A] {
    * @return
    */
   def map(f: Double ⇒ Double): A = unit(f(value))
+}
 
+abstract class StrictlyPositiveQuantity[A <: StrictlyPositiveQuantity[A]](value: Double) extends Quantity[A] {self: A ⇒
```

Comment:
@cquiroz this feature leads to many different complications that are outside of the scope of this PR, so I have decided to remove it. I have addressed all your  comments, and I think this PR is now ready to be merged! Thank you!